### PR TITLE
Disney BRDF sheen sampling improvements

### PR DESCRIPTION
**Pull request by est77**
- Use the correct sampling and pdf formulas instead of uniform hemisphere sampling.
- Improved sheen test scene.
- Updated refs images.

### CODE REVIEW

File: src/appleseed/renderer/modeling/bsdf/disneybrdf.cpp [hunk 1]
```diff
@@ -196,10 +196,22 @@ namespace
             // Compute the incoming direction in local space.
             sample.get_sampling_context().split_in_place(2, 1);
             const Vector2d s = sample.get_sampling_context().next_vector2<2>();
-            const Vector3d wi = sample_hemisphere_uniform(s);
+
+            const double phi = TwoPi * s[0];
+            const double cos_theta = 1.0 - std::pow(s[1], 1.0 / 6.0);
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/disneybrdf.cpp]: `std::` qualifiers are redundant.

File: src/appleseed/renderer/modeling/bsdf/disneybrdf.cpp [hunk 2]
```diff
@@ -614,16 +628,18 @@ namespace
...
                 if (weights[DiffuseComponent] != 0.0)
                 {
-                    pdf += DisneyDiffuseComponent().evaluate_pdf(
-                        shading_basis,
-                        incoming) * weights[DiffuseComponent];
+                    pdf +=
+                        DisneyDiffuseComponent().evaluate_pdf(
+                            shading_basis,
+                            incoming) * weights[DiffuseComponent];
                 }
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/disneybrdf.cpp]: Maybe even more readable:

    pdf +=
        weights[DiffuseComponent] *
        DisneyDiffuseComponent().evaluate_pdf(shading_basis, incoming);

File: src/appleseed/renderer/modeling/bsdf/disneybrdf.cpp [hunk 3]
```diff
@@ -196,10 +196,22 @@ namespace
             // Compute the incoming direction in local space.
             sample.get_sampling_context().split_in_place(2, 1);
             const Vector2d s = sample.get_sampling_context().next_vector2<2>();
-            const Vector3d wi = sample_hemisphere_uniform(s);
+
+            const double phi = TwoPi * s[0];
+            const double cos_theta = 1.0 - std::pow(s[1], 1.0 / 6.0);
+            const double sin_theta = std::sqrt(1.0 - square(cos_theta));
+
+            Vector3d wi =
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/disneybrdf.cpp]: Make const.

**Comment by est77** [comment]
This needs a bit more work. I'm closing it for now...

**Pull request closed without merging**